### PR TITLE
postgresql

### PR DESCRIPTION
**Pull request by Nyameliaaaa**
## whats left
- [x] remove rethink text cmds
- [x] set up postgres on server
- [x] set up postgres driver
- [x] create table schemas
- [x] rewrite commands
   - [x] rewrite econ
   - [x] rewrite shop
   - [x] rewrite cases
   - [x] rewrite profiles
      - [x] get
      - [x] set
   - [x] rewrite config
- [x] cooldown system
- [x] env update
- [x] destroy rethink package

## for kubus
- [x] destroy rethink server

### CODE REVIEW

**Comment by Nyameliaaaa** [comment]
alright gonna halt this for a bit until @SuperchupuDev finishes her client cleanup

**Comment by hkubus** [comment]
NOOOOOOOOOOOOOOOo

**Comment by Nyameliaaaa** [comment]
what

**Comment by Nyameliaaaa** [comment]
https://drawsql.app/henko-thing/diagrams/fluorine-pdf#

**Comment by hkubus** [comment]
noooooooooooooooooooooooooOOOOOOOOOOOOOOOO

File: src/classes/modules/AiModule.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import FluorineClient from './Client';
+import FluorineClient from '@classes/Client';
 import { fetch } from 'undici';
 import { Interaction, Message } from 'discord.js';
-import Embed from './Embed';
+import Embed from '@classes/Embed';
 
-export default class AI {
+export default class AiModule {
```
Comment (SuperchupuDev) [inline src/classes/modules/AiModule.ts]:
```suggestion
export default class AIModule {
```

File: src/classes/modules/PhishingModule.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    client: FluorineClient;
+    private _words: string;
+    private _users: string;
+    private _urls: string;
+
+    constructor(client: FluorineClient) {
+        this.client = client;
+        this._words = readFileSync(`${__dirname}/../../../assets/words.txt`).toString();
+        this._users = readFileSync(`${__dirname}/../../../assets/users.txt`).toString();
+        this._urls = readFileSync(`${__dirname}/../../../assets/url.txt`).toString();
```
Comment (SuperchupuDev) [inline src/classes/modules/PhishingModule.ts:20]:
```suggestion
        private this._words = readFileSync(`${__dirname}/../../../assets/words.txt`).toString();
        private this._users = readFileSync(`${__dirname}/../../../assets/users.txt`).toString();
        private this._urls = readFileSync(`${__dirname}/../../../assets/url.txt`).toString();
```

File: src/commands/config/antibot-action.ts [hunk 3]
```diff
@@ -11,8 +11,14 @@ export async function run(client: FluorineClient, interaction: CommandInteractio
             ephemeral: true
         });
     }
+
     const value = interaction.options.getString('action');
-    const guildId = interaction.guild.id;
+
+    await client.db.query<Config>('UPDATE config SET antibot_action = $1 WHERE guild_id = $2', [
+        value,
+        BigInt(interaction.guild.id)
```
Comment (SuperchupuDev) [inline src/commands/config/antibot-action.ts]:
```suggestion
        BigInt(interaction.guildId)
```

File: src/commands/config/mod-logs.ts [hunk 4]
```diff
@@ -1,24 +1,32 @@
...
             content: client.i18n.t('CONFIG_FAIL', { lng: interaction.locale }),
             ephemeral: true
         });
     }
+
     const value = interaction.options.getBoolean('mod-logs');
+
+    await client.db.query<Config>('UPDATE config SET log_moderation_actions = $1 WHERE guild_id = $2', [
+        value,
+        BigInt(interaction.guild.id)
```
Comment (SuperchupuDev) [inline src/commands/config/mod-logs.ts]:
```suggestion
        BigInt(interaction.guildId)
```

File: src/commands/config/prefix.ts [hunk 5]
```diff
@@ -12,14 +13,20 @@ export async function run(client: FluorineClient, interaction: CommandInteractio
     }
 
     const prefix = interaction.options.getString('prefix');
+
+    await client.db.query<Config>('UPDATE config SET prefix = $1 WHERE guild_id = $2', [
+        prefix,
+        BigInt(interaction.guild.id)
```
Comment (SuperchupuDev) [inline src/commands/config/prefix.ts]:
```suggestion
        BigInt(interaction.guildId)
```

File: src/commands/profile/set.ts [hunk 6]
```diff
@@ -2,14 +2,58 @@ import FluorineClient from '@classes/Client';
...
     const value = interaction.options.getString('value');
+
     switch (field) {
-        case 'location':
+        case 'birthday': {
+            const birthday = value;
+            let [day, month]: any = birthday.split('/');
+
+            day = parseInt(day) || 0;
+            month = parseInt(month) || 0;
```
Comment (SuperchupuDev) [inline src/commands/profile/set.ts]:
```suggestion
            const [day, month] = birthday.split('/').map(str => parseInt(str) || 0);
```

File: src/events/messageCreate.ts [hunk 7]
```diff
@@ -43,58 +51,41 @@ export async function run(client: FluorineClient, message: Message) {
...
-            })
-            .addLocaleField({
-                name: 'STATS_COMMAND_COUNT',
-                value: client.cmds.size.toString()
-            })
-            .addLocaleField({
-                name: 'STATS_CHANNELS_COUNT',
-                value: client.channels.cache.size.toString()
-            });
-        message.channel.send({ embeds: [embed] });
```
Comment (SuperchupuDev) [inline src/events/messageCreate.ts]: why was this deleted?

File: src/events/messageUpdate.ts [hunk 8]
```diff
@@ -1,22 +1,30 @@
...
+        ])
+    ).rows;
+
+    if (!settings.logs_enabled || !settings.logs_channel) {
         return;
     }
-    const channel = client.channels.cache.get(settings.logsChannel);
+
+    const channel = client.channels.cache.get(settings.logs_channel.toString());
     if (!channel.isText()) {
```
Comment (SuperchupuDev) [inline src/events/messageUpdate.ts]:
```suggestion
    if (!channel?.isText()) {
```
otherwise will fail if the channel doesnt exist

File: src/events/ready.ts [hunk 9]
```diff
@@ -1,17 +1,18 @@
...
-                .insert({
-                    id: g.id,
-                    prefix: process.env.DISCORD_PREFIX
-                })
-                .run(this.conn);
+        const guild = (
+            await client.db.query<Config>('SELECT logs_enabled FROM config WHERE guild_id = $1', [BigInt(g.id)])
+        ).rows;
+
+        if (guild.length === 0) {
```
Comment (SuperchupuDev) [inline src/events/ready.ts]:
```suggestion
        if (!guild.length) {
```

File: src/events/messageCreate.ts (hunk 7 above)
Comment (Nyameliaaaa) [reply src/events/messageCreate.ts]: since /stats is gone

**Review by hkubus — APPROVED** [review]
hi mom

File: src/classes/Client.ts [hunk 10]
```diff
@@ -1,21 +1,23 @@
...
 import EventHandler from '@handlers/EventHandler';
 import ApplicationCommandHandler from '@handlers/ApplicationCommandHandler';
 import CommandHandler from '@handlers/CommandHandler';
 import ComponentHandler from '@handlers/ComponentHandler';
-import EconomyHandler from '@handlers/EconomyHandler';
-import ShopHandler from '@handlers/ShopHandler';
-import TagHandler from '@handlers/TagHandler';
-import PhishingHandler from '@handlers/PhishingHandler';
+
+import AiModule from '@modules/AiModule';
```
Comment (SuperchupuDev) [inline src/classes/Client.ts]:
```suggestion
import AIModule from '@modules/AiModule';
```

File: src/classes/modules/AiModule.ts [hunk 11]
```diff
@@ -1,9 +1,9 @@
-import FluorineClient from './Client';
+import FluorineClient from '@classes/Client';
 import { fetch } from 'undici';
 import { Interaction, Message } from 'discord.js';
-import Embed from './Embed';
+import Embed from '@classes/Embed';
 
-export default class AI {
+export default class AIModule {
```
Comment (SuperchupuDev) [inline src/classes/modules/AiModule.ts]: can you rename the file to `AIModule.ts`?

File: src/classes/modules/CasesModule.ts [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+        ret.guild_id = BigInt(ret.guild_id);
+        ret.case_creator = BigInt(ret.case_creator);
+        ret.moderated_user = BigInt(ret.moderated_user);
+
+        return ret;
+    }
+
+    async getMany(guild: string, user?: User) {
+        let query;
+        let params;
```
Comment (SuperchupuDev) [inline src/classes/modules/CasesModule.ts]:
```suggestion
        let query: string;
        let params: bigint[];
```

File: src/classes/modules/CasesModule.ts [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+                    localeValue: caseObj.type.toUpperCase() as 'BAN' | 'KICK' | 'WARN' | 'MUTE' | 'TIMEOUT'
+                })
+                .addLocaleField({ name: 'CASE_MODERATOR', value: creator.tag })
+                .addLocaleField({ name: 'CASE_USER', value: member.user.tag })
+                .addLocaleField({ name: 'REASON', value: caseObj.reason })
+                .addField('ID', `#${caseObj.case_id}`);
+
+            const channel = guildObj.channels.cache.get(settings.logs_channel.toString());
+
+            if (!channel.isText()) {
```
Comment (SuperchupuDev) [inline src/classes/modules/CasesModule.ts]:
```suggestion
            if (!channel?.isText()) {
```

File: src/commands/config/currency.ts [hunk 14]
```diff
@@ -11,15 +11,21 @@ export async function run(client: FluorineClient, interaction: CommandInteractio
             ephemeral: true
         });
     }
+
     const value = interaction.options.getString('currency');
+    await client.db.query<Config>('UPDATE config SET currency = $1 WHERE guild_id = $2', [
+        value,
+        BigInt(interaction.guild.id)
```
Comment (SuperchupuDev) [inline src/commands/config/currency.ts]:
```suggestion
        BigInt(interaction.guildId)
```

File: src/commands/config/logs-channel.ts [hunk 15]
```diff
@@ -12,8 +12,14 @@ export async function run(client: FluorineClient, interaction: CommandInteractio
             ephemeral: true
         });
     }
+
     const value = interaction.options.getChannel('channel').id;
-    const guildId = interaction.guild.id;
+
+    await client.db.query<Config>('UPDATE config SET logs_channel = $1 WHERE guild_id = $2', [
+        BigInt(value),
+        BigInt(interaction.guild.id)
```
Comment (SuperchupuDev) [inline src/commands/config/logs-channel.ts]:
```suggestion
        BigInt(interaction.guildId)
```

File: src/commands/config/logs.ts [hunk 16]
```diff
@@ -1,24 +1,32 @@
...
             content: client.i18n.t('CONFIG_FAIL', { lng: interaction.locale }),
             ephemeral: true
         });
     }
+
     const value = interaction.options.getBoolean('logs');
+
+    await client.db.query<Config>('UPDATE config SET logs_enabled = $1 WHERE guild_id = $2', [
+        value,
+        BigInt(interaction.guild.id)
```
Comment (SuperchupuDev) [inline src/commands/config/logs.ts]:
```suggestion
        BigInt(interaction.guildId)
```

File: src/events/messageCreate.ts [hunk 17]
```diff
@@ -1,40 +1,48 @@
...
         return;
     }
 
-    const settings = (await r.table('config').get(message.guild?.id).run(client.conn)) as SettingsType;
-    if (settings.antibot) {
-        const factor = await messageBot(client, message);
-        if (factor >= settings.antibot) {
+    const [settings] = (
+        await client.db.query<Config>('SELECT antibot_factor, antibot_action, prefix FROM config WHERE guild_id = $1', [
+            BigInt(message.guild.id)
```
Comment (SuperchupuDev) [inline src/events/messageCreate.ts]:
```suggestion
            BigInt(message.guildId)
```

File: src/events/messageDelete.ts [hunk 18]
```diff
@@ -1,21 +1,28 @@
...
     if (!message.content) {
         return;
     }
-    // @ts-ignore
-    const settings: SettingsType = await r.table('config').get(message.guild?.id).run(client.conn);
-    if (!settings.logs || !settings.logsChannel) {
+
+    const [settings] = (
+        await client.db.query<Config>('SELECT logs_enabled, logs_channel FROM config WHERE guild_id = $1', [
+            BigInt(message.guild.id)
```
Comment (SuperchupuDev) [inline src/events/messageDelete.ts]:
```suggestion
            BigInt(message.guildId)
```

File: src/events/messageUpdate.ts [hunk 19]
```diff
@@ -1,22 +1,30 @@
...
     if (!newMessage || newMessage.content === oldMessage.content) {
         return;
     }
-    // @ts-ignore
-    const settings: SettingsType = await r.table('config').get(newMessage.guild?.id).run(client.conn);
-    if (!settings.logs || !settings.logsChannel) {
+
+    const [settings] = (
+        await client.db.query<Config>('SELECT logs_enabled, logs_channel FROM config WHERE guild_id = $1', [
+            BigInt(oldMessage.guild.id)
```
Comment (SuperchupuDev) [inline src/events/messageUpdate.ts]:
```suggestion
            BigInt(oldMessage.guildId)
```

**Review by SuperchupuDev — DISMISSED** [review]
good job!

**Review by SuperchupuDev — APPROVED** [review]
